app/code/RefreshMarkdownTask.php

```diff
@@ -1,116 +1,307 @@
 <?php
 
+/**
+ * This task will download sources defined in the $documentation_repositories and put them into the
+ * a folder where the markdown parser and indexer can work on them.
+ *
+ * Note that this downloader requires the repositories because it relies on their tarball download
+ * functionality. So this is pretty much hardcoded.
+ *
+ * The tarball downloader runs all the repos in parallel to save time and will not remove the current
+ * source until the new downloaded code can replace the old code.
+ *
+ * For local development you can use dev=1 to instead clone the repositories if you would like to
+ * work on the documentation locally instead.
+ *
+ */
 class RefreshMarkdownTask extends BuildTask
```

Comment:
Agree, I would probably just make it a repo downloader and reflect that properly in the name.